cheatsheets/gleam-for-typescript-users.md

```diff
@@ -0,0 +1,827 @@
+- [Constants](#constants)
+- [Variables type annotations](#variables-type-annotations)
+- [Data types](#data-types) -->
+  - [Strings](#strings)
+  - [Numbers](#numbers)
+  - [Arrays and Lists](#arrays-and-lists)
+  - [Tuples](#tuples)
+  - [Destructuring Arrays and Tuples](#destructuring-arrays-and-tuples)
+  - [Objects, Records, Maps](#objects-records-maps)
+- [Functions](#functions)
+- [Blocks](#blocks)
+- [Flow control](#flow-control) -->
+  - [Conditionals](#conditionals)
+  - [Switch and Case](#switch-and-case)
+  - [Piping](#piping)
+  - [Try](#try)
+- [Modules](#modules)
+
+## Comments
+
+### TypeScript
+
+In TypeScript, comments are written with a `//` prefix.
+
+```ts
+// Hello, Joe!
+```
+
+Multi-line comments may be written like so:
+
+```ts
+/*
+ * Hello, Joe!
+ */
+```
+
+TypeScript also [supports JSDoc annotations](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).
+
+```ts
+/**
+ * @type {string}
+ */
+var s;
+ 
+/** @type {Window} */
+var win;
+ 
+/** @type {PromiseLike<string>} */
+var promisedString;
+ 
+// You can specify an HTML Element with DOM properties
+/** @type {HTMLElement} */
+var myElement = document.querySelector(selector);
+element.dataset.myData = "";
+```
+
+Documentation blocks (docblocks) are extracted into generated API
+documentation.
+
+### Gleam
+
+In Gleam, comments are written with a `//` prefix.
+
+```gleam
+// Hello, Joe!
+```
+
+Comments starting with `///` are used to document the following function,
+constant, or type definition. Comments starting with `////` are used to
+document the current module.
+
+```gleam
+//// This module is very important.
+
+/// The answer to life, the universe, and everything.
+const answer: Int = 42
+
+/// A main function
+fn main() {}
+
+/// A Dog type
+type Dog {
+  Dog(name: String, cuteness: Int)
+}
+```
+
+`//` comments are not used while generating documentation files, while
+`////` and `///` will appear in them.
+## Variables
+
+You can rebind variables in both languages.
+
+### TypeScript
+
+```ts
+let a = 50;
+a = a + 100;
+a = 1;
+```
+
+### Gleam
+
+Gleam also has the `let` keyword before its variable names. However, it needs to be used explicitly each time when rebinding/reassinging a new value.
+
+```gleam
+let size = 50
+let size = size + 100
+let size = 1
+```
+## Constants
+
+### TypeScript
+
+In TypeScript, `const` is used for variables that may not be reassigned.
+
+```ts
+const example = 50;
+example = 100; // throws "TypeError: Assignment to constant variable."
+```
+
+### Gleam
+
+In Gleam constants can also be created using the `const` keyword. Constants must be literal values, defined at the top level of a module and functions cannot be used in their definitions.
+
+```gleam
+// the_question.gleam module
+const the_answer = 42
+
+pub fn main() {
+  the_answer
+}
+```
+
+They can also be marked public via the `pub` keyword and will then be
+automatically exported.
+
+```gleam
+pub const the answer = 42
+```
```

Comment:
Gleam constants are largely unrelated to TypeScript `const`. The comparison should be between in-function assignments in each language, and module level assignments in each language.